snap/pack/pack_test.go

```diff
@@ -167,6 +167,70 @@ apps:
 	c.Assert(err, ErrorMatches, `snap is unusable due to missing files: path "bin/hello-world" does not exist`)
 }
 
+func (s *packSuite) TestPackKernelGadgetOSAppWithConfigureHookHappy(c *C) {
+	for _, snapType := range []string{"kernel", "gadget", "os", "app"} {
+		snapYaml := fmt.Sprintf(`name: %[1]s
+version: 0
+type: %[1]s`, snapType)
+		sourceDir := makeExampleSnapSourceDir(c, snapYaml)
+		configureHooks := []string{"configure"}
+		c.Assert(os.Mkdir(filepath.Join(sourceDir, "meta", "hooks"), 0755), IsNil)
+		for _, hook := range configureHooks {
+			c.Assert(os.WriteFile(filepath.Join(sourceDir, "meta", "hooks", hook), []byte("#!/bin/sh"), 0755), IsNil)
+		}
+		_, err := pack.Pack(sourceDir, pack.Defaults)
+		c.Assert(err, IsNil)
+	}
+}
+
+func (s *packSuite) TestPackKernelGadgetAppWithDefaultConfigureHookHappy(c *C) {
+	for _, snapType := range []string{"kernel", "gadget", "app"} {
+		snapYaml := fmt.Sprintf(`name: %[1]s
+version: 0
+type: %[1]s`, snapType)
+		sourceDir := makeExampleSnapSourceDir(c, snapYaml)
+		configureHooks := []string{"default-configure", "configure"}
+		c.Assert(os.Mkdir(filepath.Join(sourceDir, "meta", "hooks"), 0755), IsNil)
+		for _, hook := range configureHooks {
+			c.Assert(os.WriteFile(filepath.Join(sourceDir, "meta", "hooks", hook), []byte("#!/bin/sh"), 0755), IsNil)
+		}
```

Comment:
The `configure` case is already covered in the test above, so we can remove it from this one
```suggestion
		c.Assert(os.Mkdir(filepath.Join(sourceDir, "meta", "hooks"), 0755), IsNil)
		c.Assert(os.WriteFile(filepath.Join(sourceDir, "meta", "hooks", "default-configure"), []byte("#!/bin/sh"), 0755), IsNil)
```

Reply:
`default-configure` hook always requires `configure` hook, otherwise you get an error as per `TestPackDefaultConfigureWithoutConfigureError`

Let me add a note.

Reply:
Ah of course, my mistake

Reply:
default-configure hook always requires configure hook, otherwise you get an error as per TestPackDefaultConfigureWithoutConfigureError

Will change name to TestPackKernelGadgetAppWithDefaultConfigureAndConfigureHookHappy